# ugr_ros_control Settings -----------------------
# Settings for ros_control control loop
generic_hw_control_loop:
  loop_hz: 300
  cycle_time_error_threshold: 0.01

# Settings for ros_control hardware interface
hardware_interface:
   drive_joint: axis0
   steering_joint: axis_steering
   joints:
      - axis0
      - axis_steering
   joints_config:
      axis0:
         interfaces:
            - velocity
            - effort
      axis_steering:
         interfaces:
            - velocity # No feedback here, so no state interface!

# Publish all joint states ----------------------------------
# Creates the /joint_states topic necessary in ROS
joint_state_controller:
   type:         joint_state_controller/JointStateController
   publish_rate: 50

# Joint position controller for steering
steering_position_controller:
   type: velocity_controllers/JointPositionController
   joint: axis_steering
   pid: {p: 10.0, i: 0.01, d: 0.0}

# Velocity controller for drive
drive_velocity_controller:
   type: effort_controllers/JointVelocityController
   joint: axis0
   pid: {p: 10.0, i: 0.01, d: 0.0}

drive_effort_controller:
   type: effort_controllers/JointEffortController
   joint: axis0
